saleor/payment/gateways/braintree/forms.py

```diff
@@ -30,23 +30,12 @@ def clean(self):
         # when manually adjusting the template value as we do not allow
         # partial-payments at this moment, error is returned instead.
         amount = cleaned_data.get('amount')
-        if amount and amount != self.payment.total:
+        if amount and amount != self.payment_information['amount']:
```

Comment:
Not sure of what type is `amount` here, but if it's a numeric type (which I suppose it is) and it is equal to zero, we might accidentally enter/not enter this logic, as `bool(0)` evaluates to False. Did you check that case?

Reply:
We technically have checks for that in a different place. There shouldn't be an attempt to charge 0 if it was changed via the form. The only case this is possible if the order is for 0, but that would be an okay case then.